projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -18,40 +18,42 @@ import {
 import defaultVariations from './variations';
 
 export default function JetpackPaymentsIntroEdit( { name, clientId } ) {
-	const { blockType, defaultVariation, variations, hasInnerBlocks } = useSelect( select => {
-		const { getBlockType, getBlockVariations, getDefaultBlockVariation } = select( blocksStore );
+	const { blockType, hasInnerBlocks } = useSelect( select => {
 		const { getBlocks } = select( blockEditorStore );
 
 		return {
 			blockType: getBlockType( name ),
-			defaultVariation: getDefaultBlockVariation( name, 'block' ),
-			variations: getBlockVariations( name, 'block' ),
 			hasInnerBlocks: getBlocks( clientId )?.length > 0,
 		};
 	} );
 
 	const { replaceBlock, selectBlock } = useDispatch( blockEditorStore );
 
 	const setVariation = variation => {
-		replaceBlock( clientId, createBlock( variation.name.replace( 'payments-intro', 'jetpack' ) ) );
+		replaceBlock( clientId, createBlock( variation.name ) );
 		selectBlock( clientId );
 	};
 
+	const usableVariations = unfilteredVariations => {
+		return unfilteredVariations.filter( variation => getBlockType( variation.name ) );
```

Comment:
Why do we need to filter them, to be sure that they are registered? If that's the case, maybe we should also let the developer know why their block wasn't registered in the intro?

Reply:
premium-content block is wpcom only, we don't want to show it on jetpack sites.